test/e2e/apimachinery/watchlist.go

```diff
@@ -0,0 +1,124 @@
+)
+
+var _ = SIGDescribe("API Streaming (aka. WatchList) [Serial] [Feature:WatchList]", func() {
+	f := framework.NewDefaultFramework("watchlist")
+	ginkgo.It("should be requested when ENABLE_CLIENT_GO_WATCH_LIST_ALPHA is set", func(ctx context.Context) {
+		prevWatchListEnvValue, wasWatchListEnvSet := os.LookupEnv("ENABLE_CLIENT_GO_WATCH_LIST_ALPHA")
+		os.Setenv("ENABLE_CLIENT_GO_WATCH_LIST_ALPHA", "true")
+		defer func() {
+			if !wasWatchListEnvSet {
+				os.Unsetenv("ENABLE_CLIENT_GO_WATCH_LIST_ALPHA")
+				return
+			}
+			os.Setenv("ENABLE_CLIENT_GO_WATCH_LIST_ALPHA", prevWatchListEnvValue)
+		}()
+		stopCh := make(chan struct{})
+		defer close(stopCh)
+		secretInformer := cache.NewSharedIndexInformer(
+			&cache.ListWatch{
+				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+					framework.Fail("Unexpected list call")
+					return nil, fmt.Errorf("unexpected list call")
+				},
+				WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+					return f.ClientSet.CoreV1().Secrets(f.Namespace.Name).Watch(context.TODO(), options)
+				},
+			},
+			&v1.Secret{},
+			time.Duration(0),
+			nil,
+		)
+
+		ginkgo.By(fmt.Sprintf("Adding 5 secrets to %s namespace", f.Namespace.Name))
+		for i := 1; i <= 5; i++ {
+			_, err := f.ClientSet.CoreV1().Secrets(f.Namespace.Name).Create(ctx, newSecret(fmt.Sprintf("secret-%d", i)), metav1.CreateOptions{})
+			framework.ExpectNoError(err)
+		}
+
+		ginkgo.By("Starting the secret informer")
+		go secretInformer.Run(stopCh)
+
+		ginkgo.By("Waiting until the secret informer is fully synchronised")
+		err := wait.PollUntilContextTimeout(ctx, 100*time.Millisecond, 30*time.Second, false, func(context.Context) (done bool, err error) {
+			return secretInformer.HasSynced(), nil
+		})
+		framework.ExpectNoError(err, "Failed waiting for the secret informer in %s namespace to be synced", f.Namespace.Namespace)
+
+		ginkgo.By("Verifying if the secret informer was properly synchronised")
+		verifyStore(ctx, f, secretInformer.GetStore())
+
+		ginkgo.By("Modifying a secret and checking if the update was picked up by the secret informer")
+		secret, err := f.ClientSet.CoreV1().Secrets(f.Namespace.Name).Get(ctx, "secret-1", metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		secret.StringData = map[string]string{"foo": "bar"}
+		_, err = f.ClientSet.CoreV1().Secrets(f.Namespace.Name).Update(ctx, secret, metav1.UpdateOptions{})
+		framework.ExpectNoError(err)
+		verifyStore(ctx, f, secretInformer.GetStore())
+	})
+})
+
+func verifyStore(ctx context.Context, f *framework.Framework, store cache.Store) {
+	wait.PollUntilContextTimeout(ctx, time.Second, 30*time.Second, true, func(ctx context.Context) (done bool, err error) {
+		ginkgo.By(fmt.Sprintf("Listing secrets directly from the server from %s namespace", f.Namespace.Name))
+		expectedSecretsList, err := f.ClientSet.CoreV1().Secrets(f.Namespace.Name).List(ctx, metav1.ListOptions{})
+		framework.ExpectNoError(err)
```

Comment:
Even though this will fail the test I think it is okay.

Reply:
it seems ok - I agree.